rewrite-xml/src/main/java/org/openrewrite/xml/XmlParser.java

```diff
@@ -32,9 +32,43 @@
 import org.openrewrite.xml.tree.Xml;
 
 import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.stream.Stream;
 
 public class XmlParser implements Parser {
+    private static final Set<String> ACCEPTED_FILE_EXTENSIONS = new HashSet<>(Arrays.asList(
+            "xml",
+            "wsdl",
+            "xhtml",
+            "xsd",
+            "xsl",
+            "xslt",
+            "xmi",
+            "tld",
+            "xjb",
+            "jsp",
+            // Datastage file formats that are all xml under the hood
+            "det",
+            "pjb",
+            "qjb",
+            "sjb",
+            "prt",
+            "srt",
+            "psc",
+            "ssc",
+            "tbd",
+            "tfm",
+            "dqs",
+            "stp",
+            "dcn",
+            "pst",
+            // .NET project files
+            "csproj",
+            "vbproj",
+            "fsproj"));
```

Comment:
I wonder if we also should add `"targets"` and `"props"` from the .NET ecosystem here or not. Thoughts?

Reply:
I think those were intentionally kept as plaintext, given any ambiguity in those file extensions being used elsewhere:
https://github.com/moderneinc/moderne-cli/pull/2117/files

Reply:
I see. So allowing these mappings to be more dynamic (e.g. parse `.props` files as XML in .NET projects) seems like it will at some point become necessary.